securedrop_client/storage.py

```diff
@@ -48,12 +49,13 @@ def get_local_sources(session: Session) -> List[Source]:
     return session.query(Source).all()
 
 
-def delete_local_source_by_uuid(session: Session, uuid: str) -> None:
+def delete_local_source_by_uuid(session: Session, uuid: str, data_dir: str) -> None:
     """
     Delete the source with the referenced UUID.
     """
     source = session.query(Source).filter_by(uuid=uuid).one_or_none()
     if source:
+        delete_source_collection(source.journalist_filename, data_dir)
```

Comment:
checking my understanding: before we _were_ only deleting the source locally after we successfully sent a  source-deletion job to the server, and then we'd rely on the metadata sync to tell us to delete a sources files: https://github.com/freedomofpress/securedrop-client/blob/707dde40cb55891c6c0ad2af35eda8067b54c6ab/securedrop_client/storage.py#L205-L215

this change makes it so we will delete a source's files ahead of time, before a sync? or was there a bug where the files were never getting deleted?

Reply:
Back when we were calling `sync_api` after a deletion in the deletion success callback: the sync would delete the files from a deleted source. The deletion job itself would not delete either the database row or the corresponding files, relying on the sync to do so. After we removed the `sync_api` call, it was replaced with a function in the deletion success callback that only deleted the database row. I don't think the intention was to rely on the sync in that case, since that would not have worked - the corresponding database rows (which the sync uses to determine which items to delete) are now gone.